ml-agents/mlagents/trainers/trainer.py

```diff
@@ -17,6 +19,88 @@ class UnityTrainerException(UnityException):
     pass
 
 
+class TrainerMetrics(object):
+    """
+        Helper class to track, write training metrics. Tracks time since object
+        of this class is initialized.
+    """
+
+    def __init__(self, path: str, brain_name: str):
+        """
+        :str path: Fully qualified path where CSV is stored.
+        :str brain_name: Identifier for the Brain which we are training
+        """
+        self.path = path
+        self.brain_name = brain_name
+        self.FIELD_NAMES = ['Brain name', 'Time to update policy',
+                            'Time since start of training', 'Time for last experience collection',
+                            'Number of experiences used for training', 'Mean return']
+        self.rows = []
+        self.time_start_experience_collection = None
+        self.time_training_start = time()
+        self.last_buffer_length = None
+        self.last_mean_return = None
+        self.time_policy_update_start = None
+        self.delta_last_experience_collection = None
+        self.delta_policy_update = None
+
+    def start_experience_collection_timer(self):
+        """
+        Inform Metrics class that experience collection is starting. Intended to be idempotent
+        """
+        if self.time_start_experience_collection is None:
+            self.time_start_experience_collection = time()
+
+    def end_experience_collection_timer(self):
+        """
+        Inform Metrics class that experience collection is done.
+        """
+        self.delta_last_experience_collection = time() - self.time_start_experience_collection
+        self.time_start_experience_collection = None
+
+    def start_policy_update_timer(self,  number_experiences: int, mean_return: float):
+        """
+        Inform Metrics class that policy update has started.
+        :int number_experiences: Number of experiences in Buffer at this point.
+        :float mean_return: Return averaged across all cumulative returns since last policy update
+        """
+        self.last_buffer_length = number_experiences
+        self.last_mean_return = mean_return
+        self.time_policy_update_start = time()
+
+    def end_policy_update(self):
+        """
+        Inform Metrics class that policy update has started.
+        """
+        self.delta_policy_update = time() - self.time_policy_update_start
+        delta_train_start = time() - self.time_training_start
+        logger.debug(" Policy Update Training Metrics for {}: "
```

Comment:
I would put the second part of this method (starting at this line) into a helper method.

Reply:
will be fixed